app/backend/src/couchers/models.py

```diff
@@ -654,3 +654,323 @@ class InitiatedUpload(Base):
+
+    node = relationship("Node", backref="node_cluster_associations")
+    cluster = relationship("Cluster", backref="node_cluster_associations")
+
+
+class GroupRole(enum.Enum):
+    member = 1
+    admin = 2
+
+
+class ClusterSubscription(Base):
+    """
+    ClusterSubscription of a user
+    """
+
+    __tablename__ = "cluster_subscriptions"
+    __table_args__ = (UniqueConstraint("user_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+    role = Column(Enum(GroupRole), nullable=False)
+    joined = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+    left = Column(DateTime(timezone=True), nullable=True)
+
+    user = relationship("User", backref="cluster_subscriptions")
+    cluster = relationship("Cluster", backref="cluster_subscriptions")
+
+
+class ClusterPageAssociation(Base):
+    """
+    pages related to clusters
+    """
+
+    __tablename__ = "cluster_page_associations"
+    __table_args__ = (UniqueConstraint("page_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    page_id = Column(ForeignKey("pages.id"), nullable=False, index=True)
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+
+    page = relationship("Page", backref="cluster_page_associations")
+    cluster = relationship("Cluster", backref="cluster_page_associations")
+
+
+class PageType(enum.Enum):
+    main_page = 1
+    point_of_interest = 2
+    guide = 3
+
+
+class Page(Base):
+    """
+    similar to a wiki page about a community, POI or guide
+    """
+
+    __tablename__ = "pages"
+
+    id = Column(BigInteger, primary_key=True)
+
+    type = Column(Enum(PageType), nullable=False)
+    thread_id = Column(ForeignKey("threads.id"), nullable=False, unique=True, index=True)
+    creator_user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
+    owner_user_id = Column(ForeignKey("users.id"), nullable=True, index=True)
+    owner_cluster_id = Column(ForeignKey("clusters.id"), nullable=True, unique=True, index=True)
+
+    thread = relationship("Thread", backref="page", uselist=False)
+    creator_user = relationship("User", backref="created_pages", foreign_keys="Page.creator_user_id")
+    owner_user = relationship("User", backref="owned_pages", foreign_keys="Page.owner_user_id")
+    owner_cluster = relationship("Cluster", backref="owned_page", uselist=False, foreign_keys="Page.owner_cluster_id")
+
+    editors = relationship("User", backref="editing_pages", secondary="page_versions")
+
+    # Only one of owner_user and owner_cluster should be set
+    CheckConstraint(
+        "owner_user_id IS NULL AND owner_cluster_id IS NOT NULL OR owner_user_id IS NOT NULL AND owner_cluster_id IS NULL",
+        name="one_owner",
+    )
```

Comment:
Post-merge comment: The correct syntax is

`__table_args__ = (CheckConstraint(...),)`